pkg/ruler/api.go

```diff
@@ -500,9 +500,31 @@ func (a *API) ListRules(w http.ResponseWriter, req *http.Request) {
 		return
 	}
 	if len(missing) > 0 {
-		// This API is expected to be strongly consistent, so it's an error if any rule group was missing.
-		http.Error(w, fmt.Sprintf("an error occurred while loading %d rule groups", len(missing)), http.StatusInternalServerError)
-		return
+		// This API is expected to be strongly consistent, but we expect the object storage to be strongly
+		// consistent too. This means that if a rule group existed when we listed the storage but doesn't exist
+		// (it's missing) when we load it, then it could have been deleted in the meanwhile.
+		//
+		// We don't want to consider this condition a failure, so we don't return an error in this case, but we
+		// just log a warning because it could be of interest for an operator to further investigate it.
+		level.Warn(a.logger).Log(
```

Comment:
This should be using `logger` which is a `SpanLogger`.

Reply:
Done in https://github.com/grafana/mimir/pull/9936/commits/776ffccf38dcc9c4a07a8e7bef38efc053da50df